src/qibolab/platforms/abstract.py

```diff
@@ -152,6 +163,135 @@ def reload_settings(self):
+
+    def update(self, updates: dict):
+        r"""Updates platform common runcard parameters after calibration actions.
+
+        Args:
+
+            updates (dict): Dictionary containing the parameters to update the runcard.
+        """
+
+        for par, values in updates.items():
+            for qubit, value in values.items():
+                log.info(f"qubit {qubit} - param {par} - values {value}")
+                # resonator_spectroscopy / resonator_spectroscopy_flux / resonator_punchout_attenuation
+                if par == "readout_frequency":
+                    freq = int(value * 1e9)
+                    self.native_single_qubit_gates[qubit]["MZ"]["frequency"] = freq
+                    self.current_config["native_gates"]["single_qubit"][qubit]["MZ"]["frequency"] = freq
+
+                    if "if_frequency" in self.native_single_qubit_gates[qubit]["MZ"]:
+                        self.native_single_qubit_gates[qubit]["MZ"][
+                            "if_frequency"
+                        ] = freq - self.get_lo_readout_frequency(qubit)
+                        self.current_config["native_gates"]["single_qubit"][qubit]["MZ"][
+                            "if_frequency"
+                        ] = freq - self.get_lo_readout_frequency(qubit)
+
+                    self.qubits[qubit].readout_frequency = freq
+                    self.current_config["characterization"]["single_qubit"][qubit]["readout_frequency"] = freq
+
+                # resonator_punchout_attenuation
+                elif par == "readout_attenuation":
+                    # TODO: Are we going to save the attenuation somwhere in the native_gates or characterization
+                    # in all platforms?
+                    True
+
+                # resonator_punchout_attenuation
+                elif par == "bare_resonator_frequency":
+                    freq = int(value * 1e9)
+                    self.qubits[qubit].bare_resonator_frequency = freq
+                    self.current_config["characterization"]["single_qubit"][qubit]["bare_resonator_frequency"] = freq
+
+                # resonator_spectroscopy_flux / qubit_spectroscopy_flux
+                elif par == "sweetspot":
+                    sweetspot = float(value)
+                    self.qubits[qubit].sweetspot = sweetspot
+                    self.current_config["characterization"]["single_qubit"][qubit]["sweetspot"] = sweetspot
+
+                # qubit_spectroscopy / qubit_spectroscopy_flux / ramsey
+                elif par == "drive_frequency":
+                    freq = int(value * 1e9)
+                    self.native_single_qubit_gates[qubit]["RX"]["frequency"] = freq
+                    self.current_config["native_gates"]["single_qubit"][qubit]["RX"]["frequency"] = freq
+
+                    self.qubits[qubit].drive_frequency = freq
+                    self.current_config["characterization"]["single_qubit"][qubit]["drive_frequency"] = freq
```

Comment:
Here getting even worse, need to update four different places to change a single value...

Reply:
Same as in my previous comment. But here the fitted parameter appears in 2 different places in the runcard.